addons/custom-menu-bar/username.js

```diff
@@ -0,0 +1,14 @@
+export default async function ({ addon }) {
+  if (await addon.auth.fetchIsLoggedIn()) {
+    const username = await addon.auth.fetchUsername();
+    addon.tab.redux.initialize(); // Start listening to Redux events
+    addon.tab.redux.addEventListener("statechanged", async (e) => {
+      if (!addon.settings.get("compact-username")) return; // Skip if the option is disabled
+      if (e.detail.action.type !== "scratch-gui/menus/OPEN_MENU" || e.detail.action.menu !== "accountMenu") return; // Skip if another Redux event occurs
+      const profileSpans = await addon.tab.waitForElement(".menu_menu-item_3EwYA.menu_hoverable_3u9dt"); // Do this if the menu is opened
```

Comment:
Bad idea to target a hashed class name.

Reply:
This is also happening inside the editor, so it isn't a good idea to use waitForElement lightly, without any optimizations!

Reply:
cc @DNin01 @Samq64

Reply:
I'm working on the userscript now.

Reply:
Working on a fix.

Reply:
@iqnite You'll probably want to take a quick look at #7187 before doing anything.